pkg/storage/segments/store.go

```diff
@@ -133,7 +133,17 @@ func (s *segmentStore) Put(ctx context.Context, data io.Reader, expiration time.
 			Metadata:       metadata,
 		}
 	} else {
-		limits, rootPieceID, err := s.metainfo.CreateSegment(ctx, "", "", -1, redundancy, s.maxEncryptedSegmentSize, expiration) // bucket, path and segment index are not known at this point
+		// early call to get bucket name, rest of the path cannot be determine yet
+		p, _, err := segmentInfo()
+		if err != nil {
+			return Meta{}, Error.Wrap(err)
+		}
+		bucket, _, _, err := splitPathFragments(p)
+		if err != nil {
+			return Meta{}, err
+		}
+
+		limits, rootPieceID, err := s.metainfo.CreateSegment(ctx, bucket, "", -1, redundancy, s.maxEncryptedSegmentSize, expiration) // bucket, path and segment index are not known at this point
```

Comment:
the comment at the end of this line should be removed now `// bucket, path and segment index are not known at this point`